django/core/checks/urls.py

```diff
@@ -27,12 +29,41 @@ def check_resolver(resolver):
             warnings.extend(check_resolver(pattern))
         elif isinstance(pattern, RegexURLPattern):
             warnings.extend(check_pattern_name(pattern))
+        else:
+            # This is not a url() instance
+            warnings.extend(get_warning_for_invalid_pattern(pattern))
 
-        warnings.extend(check_pattern_startswith_slash(pattern))
+        if not warnings:
+            warnings.extend(check_pattern_startswith_slash(pattern))
 
     return warnings
 
 
+def get_warning_for_invalid_pattern(pattern):
+    """
+    Return a list containing a warning that the pattern is invalid.
+
+    describe_pattern() cannot be used here, because we cannot rely on the
+    urlpattern having regex or name attributes.
+    """
+    if isinstance(pattern, six.string_types):
+        hint = (
+            "Try removing the string '{}'. The list of urlpatterns should not "
+            "have a prefix string as the first element.".format(pattern)
+        )
+    elif isinstance(pattern, tuple):
+        hint = "Try using url() instead of a tuple."
+    else:
+        hint = None
+
+    return [Warning(
```

Comment:
Maybe an error, but I'm not certain. Using a tuple or a string instead of a `url()` instance will cause an error when you try to reverse.

    >>> reverse('index')
    Traceback (most recent call last):
    ...
    AttributeError: 'tuple' object has no attribute 'regex'

So that suggests it should be an error. I don't know whether anyone would ever want to use a non `url()` object and rely on duck typing. If they are doing that, then it's probably ok to require them to silence the error.